### PR TITLE
Add imagePullSecrets as field in Collector spec

### PR DESCRIPTION
**Pull request by m1o1**
Signed-off-by: Andrew DiNunzio <<span/>adinunzio84@gmail.<span/>com>

### CODE REVIEW

**Comment by m1o1** [comment]
Some things need to be cleaned up, just need to read through the contributing docs to see what I missed.

File: bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -37,7 +37,7 @@ metadata:
     operators.operatorframework.io/project_layout: go.kubebuilder.io/v3
     repository: github.com/open-telemetry/opentelemetry-operator
     support: OpenTelemetry Community
-  name: opentelemetry-operator.v0.53.0
+  name: opentelemetry-operator.v0.53.0-4-gce77f82
```
Comment (yuriolisa) [inline bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml:40]: This file should not be changed on PR to add/fix features.

File: config/manager/kustomization.yaml [hunk 2]
```diff
@@ -1,2 +1,8 @@
 resources:
-  - manager.yaml
\ No newline at end of file
+- manager.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
```
Comment (yuriolisa) [inline config/manager/kustomization.yaml:4]: Please restore also this file.

**Review by yuriolisa — CHANGES_REQUESTED** [review]

File: config/manager/kustomization.yaml (hunk 2 above)
Comment (yuriolisa) [reply config/manager/kustomization.yaml:4]: @m1o1, could you please also sign your PR?

**Comment by m1o1** [comment]
Need to figure out how to properly set up gpg for signing, so marked as a draft for now.

**Comment by pavolloffay** [comment]
Try squshing to commits and use `-s` flag when committing (`git commit -s`).

**Comment by alupuleasa** [comment]
Do we have any updates on this? [yuriolisa](https://github.com/yuriolisa)

**Comment by yuriolisa** [comment]
> Do we have any updates on this? [yuriolisa](https://github.com/yuriolisa) can we merge this?


@alupuleasa, before we proceed with the review, @m1o1 should perform the following steps: 
- Rebase the PR due to the merge conflicts and switch the draft to PR.
- Sign the PR as expected.
- Fix the required changes.

**Comment by m1o1** [comment]
Sorry I found the opentelemetry contrib image which had the extra processor I was looking for so this fell off my radar. I'll get back to it and get it ready for review sometime this weekend.

File: config/manager/kustomization.yaml (hunk 2 above)
Comment (m1o1) [reply config/manager/kustomization.yaml]: done

**Comment by alupuleasa** [comment]
Can we proceed with the review now?

**Comment by m1o1** [comment]
apparently I'll need to fix some unit tests (could have sworn they were passing!) Will fix today

**Comment by m1o1** [comment]
Running locally, the unit tests and e2e tests are passing, but running `make test` results in a diff showing up for `bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml` and failing for that reason.

The diff is just this:
```diff
               containers:
+              - args:
+                - --secure-listen-address=0.0.0.0:8443
+                - --upstream=http://127.0.0.1:8080/
+                - --logtostderr=true
+                - --v=0
+                image: gcr.io/kubebuilder/kube-rbac-proxy:v0.11.0
+                name: kube-rbac-proxy
+                ports:
+                - containerPort: 8443
+                  name: https
+                  protocol: TCP
+                resources:
+                  limits:
+                    cpu: 500m
+                    memory: 128Mi
+                  requests:
+                    cpu: 5m
+                    memory: 64Mi
               - args:
                 - --metrics-addr=127.0.0.1:8080
                 - --enable-leader-election
@@ -311,24 +329,6 @@ spec:
                 - mountPath: /tmp/k8s-webhook-server/serving-certs
                   name: cert
                   readOnly: true
-              - args:
-                - --secure-listen-address=0.0.0.0:8443
-                - --upstream=http://127.0.0.1:8080/
-                - --logtostderr=true
-                - --v=0
-                image: gcr.io/kubebuilder/kube-rbac-proxy:v0.11.0
-                name: kube-rbac-proxy
-                ports:
-                - containerPort: 8443
-                  name: https
-                  protocol: TCP
-                resources:
-                  limits:
-                    cpu: 500m
-                    memory: 128Mi
-                  requests:
-                    cpu: 5m
-                    memory: 64Mi
               serviceAccountName: opentelemetry-operator-controller-manager
               terminationGracePeriodSeconds: 10
               volumes:
```

So it's not really different, just out of order for some reason. Otherwise, I'm not sure why the tests are failing.

File: pkg/targetallocator/deployment.go [hunk 3]
```diff
@@ -52,6 +52,7 @@ func Deployment(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTele
 					ServiceAccountName: ServiceAccountName(otelcol),
 					Containers:         []corev1.Container{Container(cfg, logger, otelcol)},
 					Volumes:            Volumes(cfg, otelcol),
+					ImagePullSecrets:   otelcol.Spec.ImagePullSecrets,
```
Comment (m1o1) [inline pkg/targetallocator/deployment.go:55]: I really have no idea how target allocators work yet, so please let me know if this should be reverted / changed

**Comment by pavolloffay** [comment]
Please also add pull secret to the statfulset.

**Comment by m1o1** [comment]
> Please also add pull secret to the statfulset.

Done

**Comment by m1o1** [comment]
I'm not sure why the security check is failing. Don't see any logs. I just pushed a change to fix the field alignment linting issue. The linter report when I run `make lint` locally shows a bunch of results not related to my changes, and `make vet` succeeds for me locally.

**Comment by m1o1** [comment]
Pulled main into my branch which I think fixes the failing e2e test, please /retest

Locally, `ensure-generate-is-noop` is still failing in `make test`, with the same diff as above. Any hints to resolve that? Thanks

File: pkg/targetallocator/deployment.go (hunk 3 above)
Comment (CoderPoet) [reply pkg/targetallocator/deployment.go:55]: Shouldn't the same field be mixed with collector? Instead use this field: `otelcol.Spec.TargetAllocator.ImagePullSecrets`

**Comment by CoderPoet** [comment]
Is the image pull secret of sidecar missing?

**Comment by yuriolisa** [comment]
@protopapa, would you like to work on that issue?

**Comment by pavolloffay** [comment]
Closing as outdatted PR. The CR has a config option to set a service account that can set the image pull secret

**Pull request closed without merging**